create.ps1

```diff
@@ -303,127 +292,126 @@ try {
-                    
-                    $outputContext.AccountReference = $responseCreateUser.id
-
-                    $outputContext.AuditLogs.Add([PSCustomObject]@{
-                            Action  = "CreateAccount" # Optionally specify a different action for this audit log
-                            Message = "Created account with username $($account.UserName)"
-                            IsError = $false
-                        })
-                    
+        if ($outputContext.AuditLogs.isError -contains $true) {
+            Throw "Error(s) occured while looking up required values"
+        }
+
+        $maxIterations = 5
+        $Iterator = 0
+        $uniqueness = $false
+
+        do {
+            $ysisAccount.'urn:ietf:params:scim:schemas:extension:ysis:2.0:User'.ysisInitials = Set-YsisInitials-Iteration -YsisInitials $($account.ysisInitials) -Iteration $Iterator
+
+            try {
+                $splatCreateUserParams = @{
+                    Uri         = "$($config.BaseUrl)/gm/api/um/scim/v2/users"
+                    Headers     = $headers
+                    Method      = 'POST'
+                    Body        = $ysisAccount | ConvertTo-Json
+                    ContentType = 'application/scim+json;charset=UTF-8'
                 }
-                catch {             
-                    $ex = $PSItem          
-                    $errorObj = Resolve-YsisV2Error -ErrorObject $ex 
-                    Write-Warning "$Iterator : $($_.Exception.Response.StatusCode) - $($errorObj.FriendlyMessage)"
-
-                    if ($_.Exception.Response.StatusCode -eq 'Conflict' -and $($errorObj.FriendlyMessage) -match "A user with the 'ysisInitials'") {                        
-                        $Iterator++
-                        Write-Warning "YSIS-Initials in use, trying with [$($account.ysisInitials)]."                            
-                    }
-                    else {
-                        throw $_                            
-                    }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    $responseCreateUser = Invoke-RestMethod @splatCreateUserParams -Verbose:$false
+                }
+                else {
+                    Write-Warning "Will send: $($splatCreateUserParams.Body)"
                 }
-            } while ($uniqueness -ne $true -and $Iterator -lt $maxIterations)
-            
-            if (!($uniqueness -eq $true)) {
-                throw "A user with the 'ysisInitials' '$($account.ysisInitials)' already exists. YSIS-Initials out of iterations!"                      
-            }            
-            if ($actionContext.DryRun -eq $true) {
-            
-                Write-Warning "Account with username [$($account.UserName)] and discipline [$($mappedObject.Discipline)] will be created."
+                $uniqueness = $true
+
+                $outputContext.AccountReference = $responseCreateUser.id
```

Comment:
This code should be moved to the DryRun not $true.

Reply:
correct. moved